lambda/translcore.ml

```diff
@@ -516,11 +555,11 @@ and transl_exp0 ~in_new_scope ~scopes e =
       let loc = of_location ~scopes e.exp_loc in
       let lam =
         match met with
-          Tmeth_val id -> Lsend (Self, Lvar id, obj, [], loc)
+          Tmeth_val id -> Lsend (Self, Lvar id, obj, [], Apply_tail, loc)
```

Comment:
It is safe to make any call/send `Apply_tail` if we know that all its parameters are `Alloc_heap`. We would like to mark `send`s as being `Apply_tail` because that allows us to continue to treat:
```ocaml
let foo o x =
  o#meth x 5 "arg"
```
as a single tail-call of the underlying method rather than as a separate `o#meth` application followed by another one that passes the rest of the args in.